pallets/ctype/src/tests.rs

```diff
@@ -25,38 +25,48 @@ use crate::{self as ctype, mock::*};
 #[test]
 fn check_successful_ctype_creation() {
 	let creator = ALICE;
-
-	let ctype_hash = get_ctype_hash::<Test>(true);
-
-	let builder = ExtBuilder::default();
-
-	let mut ext = builder.build(None);
-
-	// Write CType on chain
-	ext.execute_with(|| {
-		assert_ok!(Ctype::add(get_origin(creator.clone()), ctype_hash));
-	});
-
-	// Verify the CType has the right owner
-	let stored_ctype_creator =
-		ext.execute_with(|| Ctype::ctypes(&ctype_hash).expect("CType hash should be present on chain."));
-	assert_eq!(stored_ctype_creator, creator);
+	let ctype = [9u8; 256].to_vec();
+	let ctype_hash = <Test as frame_system::Config>::Hashing::hash(&ctype[..]);
+
+	ExtBuilder::default()
+		.with_balances(vec![(creator.clone(), <Test as ctype::Config>::Fee::get() * 2)])
+		.build()
+		.execute_with(|| {
+			assert_ok!(Ctype::add(get_origin(creator.clone()), ctype));
+			let stored_ctype_creator = Ctype::ctypes(&ctype_hash).expect("CType hash should be present on chain.");
+
+			// Verify the CType has the right owner
+			assert_eq!(stored_ctype_creator, creator);
```

Comment:
M: Add check the exact fee amount was paid.

Reply:
9961a5e